### ISSUE TITLE
Should a handler also receive sub-events when using dynamism?

### ISSUE THREAD

**Issue by almarklein** [zoofIO/flexx#120]
Consider:
```python
@connect('childen.*.mouse_down')
def _handle_child_clck(self, *events):
     for ev in events:
           ...
```
The handler will probably use `ev.source` to get the corresponding child and then do something with it. However, this handler will also get an event when the `children` property changes. Therefore it likely needs a `if ev.type == 'mouse_down'`, but this might be confusing.

On the other hand, it would make sense for this handler:
```python
@connect('childen.*.count')
def _update_total_count(self, *events):
    self.total_count = sum([child.count for chuld in children])
```

... which can probably use `@connect('children', 'childen.*.count')` in that case.

cc @Korijn

**Comment by almarklein** [comment]
BTW: this behavior probably stems from the "reactive" model that used signals instead of events.

**Comment by Korijn** [comment]
I think the distinction lies in the fact that we're subscribing to a property in one case (so we're probably defining a computed value over the set of properties) and we're listening to an interaction event in the other where the collection of emitters isn't relevant.

```python
@connect('children.*.count')
def _update_total_count(self, *events):
    self.total_count = sum([child.count for child in children])
```

If `children` changes, it makes sense that this handler is called, since the collection of `count` _properties_ we're subscribed to changes:

```python
@connect('children.*.mouse_down')
def _handle_child_clck(self, *events):
     for ev in events:
           ...
```

However, in this case, we're subscribed to a collection of `mouse_down` _event emitters_. Our handler is not a "computed value", but an interaction handler. We don't care about the collection of event handlers, we only care about the events as they are raised.

**Comment by almarklein** [comment]
Yes, but changing the behavior based on whether the event corresponds to a property or not seems like it will lead to even more confusion.

I think I like the idea that you only get events for the latest element in a 'x.y.actual_event'. It's easy and explicit to add an extra event to `@connect()`. The downside is that in a case like calculating the total count, one might forget to do that. My feeling is that the advantages outweigh this downside though. WDYT?

**Comment by Korijn** [comment]
Explicit > implicit, so I would agree with your preference in this case.

**Comment by Korijn** [comment]
Thinking about this some more; changing a collection of event emitters like `mouse_down` doesn't alter the aggregated set of values, changing a collection of properties does. Is that really so counter-intuitive? I'm trying to think of a use case where changing the behaviour for props-or-not would actually e confusing.

**Comment by almarklein** [comment]
Well, there are cases where you connect to `some_collection.*.some_prop`, and handle each event separately, using e.g. `ev.source`. If this handler *would* receive also events for changes in `some_collection`, you'd need to apply the same `if ev.type == ..` that annoyed us :)  And it would be strange (I think) if it occurred here and not for regular events. In other words, I feel uncomfortable treating events differently based on whether they correspond to a property.

**Comment by almarklein** [comment]
Closed by #130
